cpp/ql/lib/semmle/code/cpp/dataflow/internal/DataFlowImpl.qll

```diff
@@ -3031,6 +3031,17 @@ module Impl<FullStateConfigSig Config> {
       this instanceof PathNodeSinkGroup
     }
 
+    private string ppType() {
+      this instanceof PathNodeSink and result = ""
+      or
+      this.(PathNodeMid).getAp() instanceof AccessPathNil and result = ""
+      or
+      exists(DataFlowType t | t = this.(PathNodeMid).getAp().getHead().getContainerType() |
+        // The `concat` becomes "" if `ppReprType` has no result.
+        result = concat(" : " + ppReprType(t))
+      )
```

Comment:
How about
```
result = concat(" : " + ppReprType(this.(PathNodeMid).getAp().getType()))
```
where `AccessPath::getType` is implemented as `result = this.getHead().getContainerType()`?

Reply:
This is very temporary code, and `AccessPath::getType` won't be possible after https://github.com/github/codeql/pull/12930. This will be replaced by `this.(PathNodeMid).getType()`.